relay_test.go

```diff
@@ -718,3 +718,48 @@ func TestRelayThroughSeparateRelay(t *testing.T) {
 		assert.Equal(t, 1, numConns(introspected.RootPeers[serverHP]), "Expected 1 connection from relay2 to server")
 	})
 }
+
+func TestRelayConcurrentNewConnectionAttempts(t *testing.T) {
+	opts := testutils.NewOpts().SetRelayOnly()
+	testutils.WithTestServer(t, opts, func(ts *testutils.TestServer) {
+		// Create a server that is slow to accept connections by using
+		// a frame relay to slow down the initial message.
+		slowServer := testutils.NewServer(t, serviceNameOpts("slow-server"))
+		defer slowServer.Close()
+		testutils.RegisterEcho(slowServer, nil)
+
+		var delayed bool
+		relayFunc := func(outgoing bool, f *Frame) *Frame {
+			if !delayed {
+				time.Sleep(testutils.Timeout(50 * time.Millisecond))
+				delayed = true
```

Comment:
The relay function runs in two concurrent goroutines - should `delayed` be atomic?